Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/Deployable.java

```diff
@@ -23,16 +23,25 @@
 package com.microsoft.azure.hdinsight.spark.common;
 
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
 import rx.Observable;
 
+import java.io.File;
+import java.net.URI;
+
 public interface Deployable {
     /**
      * Deploy the job artifact into cluster
      *
-     * @param artifactPath the artifact to deploy
+     * @param src the artifact to deploy
      * @return Observable: upload path
      *         Observable Error: IOException;
      */
     @NotNull
-    Observable<String> deploy(@NotNull String artifactPath);
+    Observable<String> deploy(@NotNull File src, @NotNull URI dest);
+
+    @Nullable
+    default URI createUploadDir(String rootPath){
```

Comment:
No need for the method

Reply:
used to create path with uuid prefix when delpoy in SparkBatchJob

---
In reply to: [252561705](https://github.com/Microsoft/azure-tools-for-java/pull/2619#discussion_r252561705) [](ancestors = 252561705)

Reply:
Looks like you ONLY get the upload path, not create it.